Build failures on fedora when building and statically linking our own Z3

@julian-seward1 could you provide the output of running `cargo build -vv --features static-link-z3` from within the `z3-sys` directory here? Thanks!

It fails thusly:
```
     Running `/home/sewardj/MOZ/WT/RMME/z3.rs/target/debug/build/z3-sys-768ceade83050802/build-script-build`
[z3-sys 0.6.2] running: "cmake" "/home/sewardj/MOZ/WT/RMME/z3.rs/z3-sys/z3" "-DZ3_BUILD_LIBZ3_SHARED=false" "-DZ3_BUILD_EXECUTABLE=false" "-DZ3_BUILD_TEST_EXECUTABLES=false" "-DCMAKE_INSTALL_PREFIX=/home/sewardj/MOZ/WT/RMME/z3.rs/target/debug/build/z3-sys-5e134ff63c94fa4b/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/lib64/ccache/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/lib64/ccache/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/lib64/ccache/cc" "-DCMAKE_BUILD_TYPE=Debug"
[z3-sys 0.6.2] CMake Error: The source directory "/home/sewardj/MOZ/WT/RMME/z3.rs/z3-sys/z3" does not appear to contain CMakeLists.txt.
[z3-sys 0.6.2] Specify --help for usage, or press the help button on the CMake GUI.
[z3-sys 0.6.2] thread 'main' panicked at '
[z3-sys 0.6.2] command did not execute successfully, got: exit code: 1
[z3-sys 0.6.2] 
[z3-sys 0.6.2] build script failed, must exit now', /home/sewardj/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.44/src/lib.rs:885:5
[z3-sys 0.6.2] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: failed to run custom build command for `z3-sys v0.6.2 (/home/sewardj/MOZ/WT/RMME/z3.rs/z3-sys)`

Caused by:
  process didn't exit successfully: `/home/sewardj/MOZ/WT/RMME/z3.rs/target/debug/build/z3-sys-768ceade83050802/build-script-build` (exit code: 101)
  --- stdout
  running: "cmake" "/home/sewardj/MOZ/WT/RMME/z3.rs/z3-sys/z3" "-DZ3_BUILD_LIBZ3_SHARED=false" "-DZ3_BUILD_EXECUTABLE=false" "-DZ3_BUILD_TEST_EXECUTABLES=false" "-DCMAKE_INSTALL_PREFIX=/home/sewardj/MOZ/WT/RMME/z3.rs/target/debug/build/z3-sys-5e134ff63c94fa4b/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/lib64/ccache/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/lib64/ccache/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/lib64/ccache/cc" "-DCMAKE_BUILD_TYPE=Debug"

  --- stderr
  CMake Error: The source directory "/home/sewardj/MOZ/WT/RMME/z3.rs/z3-sys/z3" does not appear to contain CMakeLists.txt.
  Specify --help for usage, or press the help button on the CMake GUI.
  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', /home/sewardj/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.44/src/lib.rs:885:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

--- a/z3-sys/build.rs
+++ b/z3-sys/build.rs
@@ -54,14 +54,21 @@ fn build_z3() {
     let mut found_lib_dir = false;
     for lib_dir in &[
         "lib",
-        // Fedora builds seem to use `lib64` rather than `lib` in some
-        // circumstances.
+        // Fedora builds seem to use `lib64` rather than `lib` for 64-bit
+        // builds.
         "lib64",
     ] {
-        let lib_dir = dst.join(lib_dir);
-        if lib_dir.exists() {
+        let full_lib_dir = dst.join(lib_dir);
+        if full_lib_dir.exists() {
+            if *lib_dir == "lib64" {
+                assert_eq!(
+                    std::env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap(),
+                    "64"
+                );
+            }
+            println!("cargo:rustc-link-search=native={}", full_lib_dir.display());
             found_lib_dir = true;
-            println!("cargo:rustc-link-search=native={}", lib_dir.display(),);
+            break;
         }
     }
     assert!(

--- a/z3-sys/build.rs
+++ b/z3-sys/build.rs
@@ -51,9 +51,24 @@ fn build_z3() {
         println!("cargo:rustc-link-lib={}", cxx);
     }
 
-    let lib = dst.join("lib");
+    let mut found_lib_dir = false;
+    for lib_dir in &[
+        "lib",
+        // Fedora builds seem to use `lib64` rather than `lib` in some
+        // circumstances.
+        "lib64",
+    ] {
+        let lib_dir = dst.join(lib_dir);
+        if lib_dir.exists() {
+            found_lib_dir = true;
+            println!("cargo:rustc-link-search=native={}", lib_dir.display(),);
+        }
+    }
+    assert!(
+        found_lib_dir,
+        "Should have found the lib directory for our built Z3"
+    );
 
-    println!("cargo:rustc-link-search=native={}", lib.display());
     if cfg!(target_os = "windows") {
         println!("cargo:rustc-link-lib=static=libz3");
     } else {

